### ISSUE TITLE
Polymer 2.0 - Firebase Messaging Error - Uncaught (in promise) DOMException: Registration failed - push service error

### ISSUE THREAD

**Issue by vinvantest** [firebase/firebase-js-sdk#771]
I have a PWA app using Polymer 2.0 and firebase. 
Firebase messaging is throwing error - Uncaught (in promise) DOMException: Registration failed - push service error.

Describe your environment
 * Operating System version: Mac High Sierra 10.13.4
  * Firebase SDK - firebase@4.13.0
  * Firebase Tools version: 3.18.4
  * Firebase Product: messaging

bower polymerfire#^2.2.1        cached https://github.com/firebase/polymerfire.git#2.2.1
bower polymerfire#^2.2.1      validate 2.2.1 against https://github.com/firebase/polymerfire.git#^2.2.1
bower firebase#*                cached https://github.com/firebase/firebase-bower.git#4.6.1
bower firebase#*              validate 4.6.1 against https://github.com/firebase/firebase-bower.git#*
bower firebase#*                   new version for https://github.com/firebase/firebase-bower.git#*
bower firebase#*               resolve https://github.com/firebase/firebase-bower.git#*
bower firebase#*              download https://github.com/firebase/firebase-bower/archive/v4.13.0.tar.gz
bower firebase#*               extract archive.tar.gz
bower firebase#*              resolved https://github.com/firebase/firebase-bower.git#4.13.0
bower firebase#^4.13.0         install firebase#4.13.0
bower polymerfire#^2.2.1       install polymerfire#2.2.1

The messaging doesn't work in Chrome Desktop - [Google Chrome is up to date
Version 66.0.3359.139 (Official Build) (64-bit)] as well as in Chrome Android - [Application version Chrome 66.0.3359.126

Error trace in Chrome console Desktop version -
    Uncaught (in promise) DOMException: Registration failed - push service error
	  Promise.then (async)		
      refreshToken	@	firebase-messaging.html:87
      activateMessaging	@	firebase-messaging.html:125
      activate	@	firebase-messaging.html:243
      _bootstrapApp	@	firebase-messaging.html:259
      runMethodEffect	@	property-effects.html:813
      runEffectsForProperty	@	property-effects.html:159
      runEffects	@	property-effects.html:125
      _propertiesChanged	@	property-effects.html:1703
      _flushProperties	@	properties-changed.html:338
      _flushProperties	@	property-effects.html:1551
      _invalidateProperties	@	property-effects.html:1523
      _setProperty	@	property-effects.html:1508
      Object.defineProperty.set	@	properties-changed.html:150
      __appNameChanged	@	firebase-common-behavior.html:47
      runObserverEffect	@	property-effects.html:214
      runEffectsForProperty	@	property-effects.html:159
      runEffects	@	property-effects.html:125
      _propertiesChanged	@	property-effects.html:1703
      _flushProperties	@	properties-changed.html:338
      _flushProperties	@	property-effects.html:1551
      ready	@	property-effects.html:1656
      ready	@	element-mixin.html:546
      ready	@	dir-mixin.html:140
      ready	@	legacy-element-mixin.html:198
      ready	@	class.html:232
      ready	@	class.html:232
      ready	@	class.html:232
      _enableProperties	@	properties-changed.html:321
      connectedCallback	@	properties-mixin.html:208
      connectedCallback	@	element-mixin.html:532
      connectedCallback	@	dir-mixin.html:150
      connectedCallback	@	legacy-element-mixin.html:105
      _attachDom	@	element-mixin.html:588
      _readyClients	@	element-mixin.html:561
      _flushClients	@	property-effects.html:1565
      _propertiesChanged	@	property-effects.html:1699
      _flushProperties	@	properties-changed.html:338
      _flushProperties	@	property-effects.html:1551
      ready	@	property-effects.html:1656
      ready	@	element-mixin.html:546
      ready	@	my-app.html:540
      _enableProperties	@	properties-changed.html:321
      connectedCallback	@	properties-mixin.html:208
      connectedCallback	@	element-mixin.html:532
      (anonymous)	@	my-app.html:849

Code used 

my-app.html (PolymerElement)
    <script src="bower_components/firebase/firebase-app.js"></script>
    <script src="bower_components/firebase/firebase-auth.js"></script>
    <script src="bower_components/firebase/firebase-database.js">      
     </script>
     <script src="bower_components/firebase/firebase-firestore.js">  
     </script>
     <script src="bower_components/firebase/firebase-messaging.js">   
     </script>
     <script src="bower_components/firebase/firebase-storage.js"> 
     </script>

      <firebase-app
        auth-domain="[[configFirebase.authDomain]]"
        database-url="[[configFirebase.databaseURL]]"
        api-key="[[configFirebase.apiKey]]"
        storage-bucket="[[configFirebase.storageBucket]]"
        messaging-sender-id="[[configFirebase.messagingSenderId]]">
      </firebase-app>

      <firebase-auth
        id="auth"
        user="{{user}}"
        signed-in="{{signedIn}}"
        status-known="{{statusKnown}}"
        on-error="_handleError">
      </firebase-auth>

      <firebase-messaging
            id="messaging"
            token="{{token}}"
            on-message="handleMessage"
            active="{{fcmActive}}"
            status-known="{{fcmStatusKnown}}">
      </firebase-messaging>

       <firebase-document path="/users/[[user.uid]]/token" data=". 
        {{tokenDocVal}}">
       </firebase-document>

      class MyApp extends Polymer.Element {
           static get is() { return 'my-app'; }

        ready() {
          this.addEventListener('request-fcm-permission',        
             this._requestFcmPermission);
          super.ready();
        }

       _requestFcmPermission() {
          var fcm = this.$.messaging;
          var self = this;
          if (this.user && !fcm.active) {
            fcm.requestPermission()
              .then(function(){
                console.log('FCM permission granted');
                console.log('[[token]] ->' + self.token);
                self.tokenDocVal = self.token;
              })
              .catch(function(e){
                console.warn('FCM permission blocked', e);
              })
          }
        }

        ... other html code ....

And the 
        firebase-messaging-sw.js 

        importScripts('/bower_components/firebase/firebase-app.js');
        importScripts('/bower_components/firebase/firebase-
           messaging.js');

       firebase.initializeApp({
        'messagingSenderId': '<project-ms-id>'
       });

       var messaging = firebase.messaging();

       messaging.setBackgroundMessageHandler(function(payload) {
       console.log('[firebase-messaging-sw.js] Received background 
         message 
          ', payload);

       var notificationTitle = payload.notification.title;
       var notificationOptions = {
       body: payload.notification.body,
       icon: '../images/fresh/android/android-launchericon-24-24.png'
        };
       return self.registration.showNotification(notificationTitle,
        notificationOptions);
       });

       Manifest.json includes "gcm_sender_id": "103953800507",

Request help to resolve the issue. I have been on this issue for long and don't know why its happening and how to fix it.

I have also added a comment on the issue on GITHUB -    
     https://github.com/firebase/firebase-js-sdk/issues/152

**Comment by mmermerkaya** [comment]
Firebase SDK does not have a 3.18.4 version. See versions on NPM [here](https://www.npmjs.com/package/firebase?activeTab=versions).

Please let us know the actual SDK version you're using.

**Comment by vinvantest** [comment]
My bad. The Firebase SDK version is - firebase@4.13.1

**Comment by vinvantest** [comment]
Hi, any luck in resolving or guidance on where it is going wrong.

**Comment by mmermerkaya** [comment]
I can't reproduce this. Do you have an online app that I can test? Or somewhere I can see your whole project?

**Comment by mmermerkaya** [comment]
Closing for now, will reopen if you can help us reproduce the issue.

**Comment by vinvantest** [comment]
No ... I haven't done it in firebase/polymerfire yet. I will raise it now and provide a link here.
https://github.com/firebase/polymerfire/issues/345

**Comment by vinvantest** [comment]
Hi... I raised a support ticket with Firebase... I suggested to raise it with Polymerfire team.

On 7 Jun 2018, at 6:58 am, firebase-support@google.com wrote:

Hello Vinayak,

If this is an issue with PolymerFire, you'll need to follow up on their issue tracker. I also suggest checking out some of the solutions from StackOverflow. It looks like, for most people, the error is happening due to some issue with the manifest file.

Regards,
Andy

I checked my manifest file and it is getting loaded correctly and has correct MIME/Type: application/json

Not sure how to resolve the issue.

Manifest file -->
{
    "name": "<app_name>",
    "short_name": "<app_name_short>",
    "description": "<desc>",
    "start_url": "./?utm_source=web_app_manifest",
    "scope": "./",
    "display": "standalone",
    "theme_color": "#fff",
    "background_color": "#fff",
    "gcm_sender_id": "103953800507",
}
